CMakeLists.txt

```diff
@@ -146,6 +146,21 @@ elseif (TARGET_PLATFORM STREQUAL "linux")
     set(BEEROCKS_BH_WIRE_IFACE "eth0")
 endif()
 
+# Test support
+option(BUILD_TESTS "build multiap unit tests" OFF)
+if (BUILD_TESTS)
+    enable_testing()
+    message(STATUS "Tests are enabled")
+    # GTest is optional. gtest-based test have to check GTest_FOUND (which implies BUILD_TESTS)
+    # and link with GTest::gtest GTest::gtest_main GTest::gmock GTest::gmock_main as appropriate
+    # CMake-bundled FindGTest doesn't support gmock, so force the one installed by GTest itself.
+    find_package(GTest CONFIG)
```

Comment:
Maybe it's just a typo, but even if googletest has been installed in docker image, this results in "Could NOT find GTest (missing: GTest_DIR)"

googlemock is installed together with googletest in /usr/src/googletest so I think that if GTest is found, so will be GMock. GoogleTest and GoogleMock used to be released as different projects, but they are so closely related that maintainers decided to merge them into the same repository.

Reply:
> Maybe it's just a typo, but even if googletest has been installed in docker image, this results in "Could NOT find GTest (missing: GTest_DIR)"

That's because it actually isn't found - the Ubuntu package only contains the source, not the ("compiled") CMake support files. I apparently only tested on my Fedora.